dashboard/modules/job/tests/test_cli_integration.py

```diff
@@ -278,22 +278,24 @@ def test_quote_escaping(ray_start_stop):
 
 
 def test_resources(shutdown_only):
-    ray.init(num_cpus=1, num_gpus=1, resources={"Custom": 1})
+    ray.init(num_cpus=1, num_gpus=1, resources={"Custom": 1}, _memory=4)
 
     # Check the case of too many resources.
     for id, arg in [
         ("entrypoint_num_cpus", "--entrypoint-num-cpus=2"),
         ("entrypoint_num_gpus", "--entrypoint-num-gpus=2"),
+        ("entrypoint_memory", "--entrypoint-memory=4"),
         ("entrypoint_resources", "--entrypoint-resources='{\"Custom\": 2}'"),
     ]:
         _run_cmd(f"ray job submit --submission-id={id} --no-wait {arg} -- echo hi")
         stdout, _ = _run_cmd(f"ray job status {id}")
+        print(f"stdout: {stdout}")
         assert "waiting for resources" in stdout
```

Comment:
@architkulkarni I can't seem to force the job into `PENDING` state to print out this line:
https://github.com/ray-project/ray/blob/f2eaea027f3744c3008c1a6b7ed6553c46de250b/dashboard/modules/job/common.py#L118

In CI the job seem to succeed?
```
________________________________ test_resources ________________________________
--
  |  
  | shutdown_only = None
  |  
  | def test_resources(shutdown_only):
  | ray.init(num_cpus=1, num_gpus=1, resources={"Custom": 1}, _memory=4)
  |  
  | # Check the case of too many resources.
  | for id, arg in [
  | ("entrypoint_num_cpus", "--entrypoint-num-cpus=2"),
  | ("entrypoint_num_gpus", "--entrypoint-num-gpus=2"),
  | ("entrypoint_memory", "--entrypoint-memory=4"),
  | ("entrypoint_resources", "--entrypoint-resources='{\"Custom\": 2}'"),
  | ]:
  | _run_cmd(f"ray job submit --submission-id={id} --no-wait {arg} -- echo hi")
  | stdout, _ = _run_cmd(f"ray job status {id}")
  | >           assert "waiting for resources" in stdout or "is currently running" in stdout
  | E           assert ('waiting for resources' in "2023-09-24 23:39:39,228\tINFO cli.py:36 -- \x1b[37mJob submission server address\x1b[39m: \x1b[1mhttp://127.0.0.1:8265\x1b[22m\n2023-09-24 23:39:39,232\tSUCC cli.py:60 -- \x1b[32m---------------------------------\x1b[39m\n2023-09-24 23:39:39,232\tSUCC cli.py:61 -- \x1b[32mJob 'entrypoint_memory' succeeded\x1b[39m\n2023-09-24 23:39:39,232\tSUCC cli.py:62 -- \x1b[32m---------------------------------\x1b[39m\n" or 'is currently running' in "2023-09-24 23:39:39,228\tINFO cli.py:36 -- \x1b[37mJob submission server address\x1b[39m: \x1b[1mhttp://127.0.0.1:8265\x1b[22m\n2023-09-24 23:39:39,232\tSUCC cli.py:60 -- \x1b[32m---------------------------------\x1b[39m\n2023-09-24 23:39:39,232\tSUCC cli.py:61 -- \x1b[32mJob 'entrypoint_memory' succeeded\x1b[39m\n2023-09-24 23:39:39,232\tSUCC cli.py:62 -- \x1b[32m---------------------------------\x1b[39m\n")
  |  
  | python/ray/dashboard/modules/job/tests/test_cli_integration.py:292: AssertionError
  | ----------------------------- Captured stdout call -----------------------------
  | Running command: 'ray job submit --submission-id=entrypoint_num_cpus --no-wait --entrypoint-num-cpus=2 -- echo hi'
  | Command succeeded.
  | Running command: 'ray job status entrypoint_num_cpus'
  | Command succeeded.
  | Running command: 'ray job submit --submission-id=entrypoint_num_gpus --no-wait --entrypoint-num-gpus=2 -- echo hi'
  | Command succeeded.
  | Running command: 'ray job status entrypoint_num_gpus'
  | (autoscaler +4m45s) Tip: use `ray status` to view detailed cluster status. To disable these messages, set RAY_SCHEDULER_EVENTS=0.
  | (autoscaler +4m45s) Error: No available node types can fulfill resource request {'GPU': 2.0}. Add suitable node types to this cluster to resolve this issue.
  | (autoscaler +4m45s) Error: No available node types can fulfill resource request {'CPU': 2.0}. Add suitable node types to this cluster to resolve this issue.
  | Command succeeded.
  | Running command: 'ray job submit --submission-id=entrypoint_memory --no-wait --entrypoint-memory=4 -- echo hi'
  | Command succeeded.
  | Running command: 'ray job status entrypoint_memory'
  | Command succeeded.
  | ----------------------------- Captured stderr call -----------------------------
  | 2023-09-24 23:39:26,798	WARNING services.py:1889 -- WARNING: The object store is using /tmp instead of /dev/shm because /dev/shm has only 2684354560 bytes available. This will harm performance! You may be able to free up space by deleting files in /dev/shm. If you are inside a Docker container, you can increase /dev/shm size by passing '--shm-size=4.45gb' to 'docker run' (or add it to the run_options list in a Ray cluster config). Make sure to set this to more than 30% of available RAM.
  | 2023-09-24 23:39:26,911	INFO worker.py:1665 -- Started a local Ray instance. View the dashboard at http://127.0.0.1:8265
  | =========================== short test summary info ============================
  | FAILED python/ray/dashboard/modules/job/tests/test_cli_integration.py::test_resources
  | =================== 1 failed, 21 passed in 289.52s (0:04:49) ===================
```

When I run this test locally, the job status is `RUNNING`:
```sh
            print(f"stdout: {stdout}")
>           assert "waiting for resources" in stdout
E           assert 'waiting for resources' in "2023-09-25 08:00:33,237\tINFO cli.py:36 -- \x1b[37mJob submission server address\x1b[39m: \x1b[1mhttp://127.0.0.1:826...b 'entrypoint_memory': RUNNING\n2023-09-25 08:00:33,240\tINFO cli.py:88 -- Status message: Job is currently running.\n"

python/ray/dashboard/modules/job/tests/test_cli_integration.py:293: AssertionError
-------------------------------------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------------------------------------
Usage stats collection is enabled. To disable this, run the following command: `ray disable-usage-stats` before starting Ray. See https://docs.ray.io/en/master/cluster/usage-stats.html for more details.
Running command: 'ray job submit --submission-id=entrypoint_num_cpus --no-wait --entrypoint-num-cpus=2 -- echo hi'
Command succeeded.
Running command: 'ray job status entrypoint_num_cpus'
Command succeeded.
stdout: 2023-09-25 08:00:28,409	INFO cli.py:36 -- Job submission server address: http://127.0.0.1:8265
2023-09-25 08:00:29,423	INFO cli.py:86 -- Status for job 'entrypoint_num_cpus': PENDING
2023-09-25 08:00:29,423	INFO cli.py:88 -- Status message: Job has not started yet. It may be waiting for resources (CPUs, GPUs, memory, custom resources) to become available.

Running command: 'ray job submit --submission-id=entrypoint_num_gpus --no-wait --entrypoint-num-gpus=2 -- echo hi'
Command succeeded.
Running command: 'ray job status entrypoint_num_gpus'
Command succeeded.
stdout: 2023-09-25 08:00:31,366	INFO cli.py:36 -- Job submission server address: http://127.0.0.1:8265
2023-09-25 08:00:31,369	INFO cli.py:86 -- Status for job 'entrypoint_num_gpus': PENDING
2023-09-25 08:00:31,369	INFO cli.py:88 -- Status message: Job has not started yet. It may be waiting for resources (CPUs, GPUs, memory, custom resources) to become available.

Running command: 'ray job submit --submission-id=entrypoint_memory --no-wait --entrypoint-memory=4 -- echo hi'
(autoscaler +11s) Tip: use `ray status` to view detailed cluster status. To disable these messages, set RAY_SCHEDULER_EVENTS=0.
(autoscaler +11s) Error: No available node types can fulfill resource request {'GPU': 2.0}. Add suitable node types to this cluster to resolve this issue.
(autoscaler +11s) Error: No available node types can fulfill resource request {'CPU': 2.0}. Add suitable node types to this cluster to resolve this issue.
Command succeeded.
Running command: 'ray job status entrypoint_memory'
Command succeeded.
stdout: 2023-09-25 08:00:33,237	INFO cli.py:36 -- Job submission server address: http://127.0.0.1:8265
2023-09-25 08:00:33,240	INFO cli.py:86 -- Status for job 'entrypoint_memory': RUNNING
2023-09-25 08:00:33,240	INFO cli.py:88 -- Status message: Job is currently running.

-------------------------------------------------------------------------------------------------------- Captured stderr call ---------------------------------------------------------------------------------------------------------
2023-09-25 08:00:22,274	INFO worker.py:1665 -- Started a local Ray instance. View the dashboard at 127.0.0.1:8265 
```

From above we see `stdout` printed out:
```
Running command: 'ray job status entrypoint_memory'
Command succeeded.
stdout: 2023-09-25 08:00:33,237	INFO cli.py:36 -- Job submission server address: http://127.0.0.1:8265
2023-09-25 08:00:33,240	INFO cli.py:86 -- Status for job 'entrypoint_memory': RUNNING
2023-09-25 08:00:33,240	INFO cli.py:88 -- Status message: Job is currently running.
```

Do you have any suggestions?

Reply:
My guess is the resource request of 4 bytes was available, so the job was successfully scheduled.  Can you try with a much larger memory?  Ideally, we can get the available memory from a Ray API, and then double it, instead of choosing an arbitrary large value.

Reply:
Ah it looks like you specified 4 bytes of memory in the `ray.init()` call.  In that case, try passing anything larger than 4 in the job. Ideally we could name it as a constant to make it clear. `available_memory = 4`, and then pass `available_memory + 1` to the job

Reply:
That worked! Thanks!

Reply:
@architkulkarni seems like sometimes this test would time out. Is that normal?